### PR TITLE
changed _departed_ids, and _arrived_ids in the update function

### PR DESCRIPTION
**Pull request by Yasharzf**
## Pull request information

- **Status**:  ready to merge
- **Kind of changes**: bug fix 
- **Related PR or issue**: 

## Description

In vehicle/traci, we are storing all departed and arrived vehicle IDs for each simulation step. Every simulation step we are appending new vehicle list to _departed_ids and _arrived_ids. In large networks or high horizon times, this can be expensive and results in significant slow down. I changed _departed_ids and _arrived_ids to just store the current ids. I didn't totally delete them since we may need them (e.g., as in MultiEnv)

### CODE REVIEW

File: flow/core/kernel/vehicle/traci.py [hunk 1]
```diff
@@ -518,7 +517,7 @@ def get_num_arrived(self):
     def get_arrived_ids(self):
         """See parent class."""
         if len(self._arrived_ids) > 0:
-            return self._arrived_ids[-1]
+            return self._arrived_ids
```
Comment (eugenevinitsky) [inline flow/core/kernel/vehicle/traci.py:520]: why this change? this affects a lot of the codebase. Otherwise LGTM

File: flow/core/kernel/vehicle/traci.py [hunk 2]
```diff
@@ -532,7 +531,7 @@ def get_arrived_rl_ids(self):
     def get_departed_ids(self):
         """See parent class."""
         if len(self._departed_ids) > 0:
-            return self._departed_ids[-1]
+            return self._departed_ids
```
Comment (eugenevinitsky) [inline flow/core/kernel/vehicle/traci.py:534]: same here

File: flow/core/kernel/vehicle/traci.py (hunk 1 above)
Comment (Yasharzf) [reply flow/core/kernel/vehicle/traci.py:520]: self._arrived_ids was a list storing ids for each step. in this function we just want the ids in the last step ([-1]). Since I don't have a list anymore, I deleted [-1].

Comment (eugenevinitsky) [reply flow/core/kernel/vehicle/traci.py:520]: So if it's not a list anymore why does it have a len method?

Comment (Yasharzf) [reply flow/core/kernel/vehicle/traci.py:520]: good catch!

File: flow/core/kernel/vehicle/traci.py [hunk 3]
```diff
@@ -211,11 +211,10 @@ def update(self, reset):
                     self.__vehicles[veh_id]["last_lc"] = self.time_counter
 
             # updated the list of departed and arrived vehicles
-            self._num_departed.append(
-                len(sim_obs[tc.VAR_DEPARTED_VEHICLES_IDS]))
-            self._num_arrived.append(len(sim_obs[tc.VAR_ARRIVED_VEHICLES_IDS]))
-            self._departed_ids.append(sim_obs[tc.VAR_DEPARTED_VEHICLES_IDS])
-            self._arrived_ids.append(sim_obs[tc.VAR_ARRIVED_VEHICLES_IDS])
+            self._num_departed.append(sim_obs[tc.VAR_LOADED_VEHICLES_NUMBER])
+            self._num_arrived.append(sim_obs[tc.VAR_ARRIVED_VEHICLES_NUMBER])
```
Comment (eugenevinitsky) [inline flow/core/kernel/vehicle/traci.py:215]: as long as there is a getter for these, this LGTM
